### PR TITLE
Codify cluster id in stats filter.

### PR DESCRIPTION
**Pull request by bianpengyuan**
Since now we unconditionally add cluster labels with EnvoyFilter configuration, it is better to just codify them within the stats filter. Corresponding istio PR: https://github.com/istio/istio/pull/31226

Also make a fix for https://github.com/istio/istio/issues/29373. Corresponding istio PR: https://github.com/istio/istio/pull/31214

### CODE REVIEW

File: extensions/common/context.cc [hunk 1]
```diff
@@ -316,11 +317,19 @@ bool extractPeerMetadataFromUpstreamHostMetadata(
...
+    // name if there are the fifth part.
+    cluster_id = fbb.CreateString(parts[4]);
+  }
   auto labels_offset = fbb.CreateVectorOfSortedTables(&labels);
 
   FlatNodeBuilder node(fbb);
   node.add_workload_name(workload_name);
   node.add_namespace_(namespace_);
+  if (!cluster_id.IsNull()) {
+    node.add_cluster_id(cluster_id);
```
Comment (douglas-reid) [inline extensions/common/context.cc:331]: qq: do we need to default this to something if cluster id is null (like "kubernetes" or "unknown")?

Comment (bianpengyuan) [reply extensions/common/context.cc:331]: Currently, with Envoyfilter config, I think the cluster labels will just be `unknown` when it is unknown, and stats filter will fill in unknown for empty labels by default. For example in one of the test, you can find that source cluster is set to unknown when mx is not happening:
```
  - name: source_cluster
    value: unknown
```

Comment (douglas-reid) [reply extensions/common/context.cc:331]: ok. as long as stats_filter continues to populate `unknown` for missing values, we should be good.

File: testdata/bootstrap/client.yaml.tmpl [hunk 2]
```diff
@@ -60,7 +60,7 @@ static_resources:
           metadata:
             filter_metadata:
               istio:
-                workload: ratings-v1;default;ratings;version-1
+                workload: ratings-v1;default;ratings;version-1;test-cluster
```
Comment (kyessenov) [inline testdata/bootstrap/client.yaml.tmpl:63]: nit: would be good to have a different value for destination cluster than source.

**Comment by kyessenov** [comment]
/lgtm
/approve
Nit to improve the test, DNM if Doug has more concerns.

**Review by kyessenov — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]

File: testdata/bootstrap/client.yaml.tmpl (hunk 2 above)
Comment (bianpengyuan) [reply testdata/bootstrap/client.yaml.tmpl]: Done.

**Pull request merged**